### PR TITLE
Document supported SemVer ranges with example

### PR DESCRIPTION
**Pull request by DanielRuf**
Let's document this. Reading the code does not replace a good documentation.

See https://github.com/actions/setup-node/issues/61

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,12 @@ This action sets by node environment for use in actions by:
 
 See [action.yml](action.yml)
 
+You can use any valid [SemVer](https://semver.org/) range supported by the [sem-ver](https://www.npmjs.com/package/semver#ranges) module.
+
+For example [`node-version: '*'`](https://www.npmjs.com/package/semver#x-ranges-12x-1x-12-) would satisfy the latest version.
+
+See also the [`evaluateVersions`](https://github.com/actions/setup-node/blob/93313caf82afbdb49b8e4790ac7f1c97c8034826/src/installer.ts#L114) method and its usage of [`semver.satisfies`](https://github.com/actions/setup-node/blob/93313caf82afbdb49b8e4790ac7f1c97c8034826/src/installer.ts#L130)
```
Comment (bryanmacfarlane) [inline README.md:20]: would prefer to keep links to code (which goes stale over time) out of the main readme.  this is more of why the feature works.  I think it's enough to just say the preceeding points about any semver is supported.  so would prefer to remove this paragraph

File: README.md [hunk 2]
```diff
@@ -13,6 +13,12 @@ This action sets by node environment for use in actions by:
 
 See [action.yml](action.yml)
 
+You can use any valid [SemVer](https://semver.org/) range supported by the [sem-ver](https://www.npmjs.com/package/semver#ranges) module.
+
+For example [`node-version: '*'`](https://www.npmjs.com/package/semver#x-ranges-12x-1x-12-) would satisfy the latest version.
+
```
Comment (bryanmacfarlane) [inline README.md:19]: can point here for more examples: https://github.com/actions/setup-node/blob/main/.github/workflows/versions.yml

**Comment by bryanmacfarlane** [comment]
any interest in updating to target main instead of master?

**Comment by DanielRuf** [comment]
> any interest in updating to target main instead of master?

Done. Normally you should also be able to do that.
Since there are conflicts and this is probably completely out of date, I will close this PR.

**Pull request closed without merging**